Sources/MaliciousSiteProtection/Services/FileStore.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  FileStore.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import os
+
+public protocol FileStoring {
+    func write(data: Data, to filename: String)
+    func read(from filename: String) -> Data?
+}
+
+public struct FileStore: FileStoring {
+    private let dataStoreURL: URL
+
+    public init() {
+        let dataStoreDirectory: URL
+        do {
+            dataStoreDirectory = try FileManager.default.url(for: .applicationSupportDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
```

Comment:
Again, doesn't affect this PR but just making notes for myself: in the next PR we should try to align with @alessandroboron's TD for storage locations: https://app.asana.com/0/481882893211075/1207273224076495/f

Reply:
this part is not modified from the original implementation: to be update in later PRs to match the config files storage location